### ISSUE TITLE
Escaping nested routing when pages are split into different layers

### ISSUE THREAD

**Issue by wonszfill** [nuxt/nuxt#21787]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.15.0`
- Nuxt Version:     `3.5.1`
- Nitro Version:    `2.4.1`
- Package Manager:  `pnpm@8.5.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://github.com/wonszfill/nuxtLayersNestedRoutes

Repository containing example with three layers
- front (main layer)
- orders
- contractors
The repository contains two branches, showing difference in behaviour of nested routing when pages are split into layers.

### Describe the bug

When routing is splitted into layers it is possible to escape nested routing.

Page contractorId is supposed to have some nested routes -> conractors profile, data etc. These pages should have common menu set up in [contractorId]. On the other hand there happens to be [officeId] page, which must not have said common menu from [contractorId] nested routing.

When these pages are set up in a way where [contractorId] and desired nested routes are in separate layer from [officeId] in works in that way - content from [contractorId] is not visible in [officeId].

Layers documentation does not cover this topic (nested pages). As for my use case, behaviour described above could be beneficial, but I understand that it could be considered as bug and fixed in the future, impacting my current routing.

**Comment by harlan-zw** [comment]
Hey @wonszfill, thanks for the issue and reproduction repo!

It's hard to follow exactly what the issue is you're describing. Do you think you could also make a minimal reproduction highlighting the issue in StackBlitz? :pray: 

Any reduced complexity would be a plus

**Comment by slarrasoain** [comment]
@harlan-zw I think it is the same bug that i describe and make a minimal repro in this issue : #15687

**Comment by danielroe** [comment]
Okay, let's track there.
